### PR TITLE
Almedina ms/nav panel

### PR DESCRIPTION
**Pull request by almedina-ms**
submenus

### CODE REVIEW

File: src/TerminalSettings/TerminalSettings/MainPage.cpp [hunk 1]
```diff
@@ -56,39 +56,39 @@ namespace winrt::TerminalSettings::implementation
 
     }
 
-    void MainPage::SettingsNav_ItemInvoked(Controls::NavigationView sender, Controls::NavigationViewItemInvokedEventArgs args)
+    void winrt::TerminalSettings::implementation::MainPage::SettingsNav_SelectionChanged_1(winrt::Microsoft::UI::Xaml::Controls::NavigationView const& sender, winrt::Microsoft::UI::Xaml::Controls::NavigationViewSelectionChangedEventArgs const& args)
```
Comment (carlos-zamora) [inline src/TerminalSettings/TerminalSettings/MainPage.cpp]:
```suggestion
    void MainPage::SettingsNav_SelectionChanged_1(NavigationView const& sender, NavigationViewSelectionChangedEventArgs const& args)
```
And on line 10, replace
```c++
using namespace Windows::UI::Xaml;
```
with
```c++
using namespace winrt::Microsoft::UI::Xaml::Controls;
using namespace winrt::Windows::UI::Xaml;
```

File: src/TerminalSettings/TerminalSettings/MainPage.cpp [hunk 2]
```diff
@@ -56,39 +56,39 @@ namespace winrt::TerminalSettings::implementation
...
+        if (clickedItemContainer != NULL)
         {
-            const hstring globalsPage = L"Globals_Page";
-            const hstring profilesPage = L"Profiles_Page";
-            const hstring colorSchemesPage = L"ColorSchemes_Page";
-            const hstring keybindingsPage = L"Keybindings_Page";
+            const hstring globalsPage = L"General_Nav";
+            const hstring profilesPage = L"Profiles_Nav";
+            const hstring colorSchemesPage = L"Appearance_Nav";
+            const hstring keybindingsPage = L"Keyboard_Nav";
```
Comment (carlos-zamora) [inline src/TerminalSettings/TerminalSettings/MainPage.cpp:60]: We could probably make these `constexpr` at the top of the file.

File: src/TerminalSettings/TerminalSettings/MainPage.h [hunk 3]
```diff
@@ -16,6 +16,9 @@ namespace winrt::TerminalSettings::implementation
         void SettingsNav_Loaded(Windows::Foundation::IInspectable const& sender, Windows::UI::Xaml::RoutedEventArgs const& args);
         void SettingsNav_SelectionChanged(Windows::UI::Xaml::Controls::NavigationView sender, Windows::UI::Xaml::Controls::NavigationViewSelectionChangedEventArgs args);
         void SettingsNav_ItemInvoked(Windows::UI::Xaml::Controls::NavigationView sender, Windows::UI::Xaml::Controls::NavigationViewItemInvokedEventArgs args);
+        void SettingsNav_Loaded_1(Windows::Foundation::IInspectable const& sender, Windows::UI::Xaml::RoutedEventArgs const& e);
+        void SettingsNav_SelectionChanged_1(Microsoft::UI::Xaml::Controls::NavigationView const& sender, Microsoft::UI::Xaml::Controls::NavigationViewSelectionChangedEventArgs const& args);
+        void SettingsNav_ItemInvoked_1(Microsoft::UI::Xaml::Controls::NavigationView const& sender, Microsoft::UI::Xaml::Controls::NavigationViewItemInvokedEventArgs const& args);
```
Comment (carlos-zamora) [inline src/TerminalSettings/TerminalSettings/MainPage.h]: You probably don't need to append the `_1`.

Comment (carlos-zamora) [reply src/TerminalSettings/TerminalSettings/MainPage.h]: You'd have to propagate those changes to the MainPage.xaml and MainPage.cpp too

File: src/TerminalSettings/TerminalSettings/MainPage.xaml [hunk 4]
```diff
@@ -8,68 +8,150 @@
...
-            <NavigationViewItem>
-                <NavigationViewItem.Icon>
+                </muxc:NavigationViewItem.Icon>
+                <TextBlock Tag="CmdPalette_Page">Command Palette</TextBlock>
+            </muxc:NavigationViewItem>
+
+            <muxc:NavigationViewItem Icon="Shop"
+                                    Tag="Marketplace_Nav">
+                <TextBlock Tag="Marketplace_Page">Marketplace</TextBlock>
+            </muxc:NavigationViewItem>
```
Comment (carlos-zamora) [inline src/TerminalSettings/TerminalSettings/MainPage.xaml]: You can remove all of these. We're not gonna have the Mouse, Command Palette, and Marketplace pages yet.

Comment (vineeththomasalex) [reply src/TerminalSettings/TerminalSettings/MainPage.xaml]: image

**Comment by vineeththomasalex** [comment]
We'll do a final run through on Wednesday to Polish the code and fix PR comments.

**Review by carlos-zamora — APPROVED** [review]

**Pull request merged**